### PR TITLE
Fix startup cleanup guard and calibration exclusions

### LINKED ISSUE #185: Session-cleanup-detector: calibration data false positives + redundant pipeline execution
## Summary

Two related issues with `session-cleanup-detector.ps1`:

### 1. Calibration data flagged as stale tracking artifacts

The cleanup detector recursively scans all files in `.copilot-tracking/` looking for `issue_id:` YAML fields. Files without that field are flagged as "unknown" stale artifacts. The calibration data introduced by #141 (`.copilot-tracking/calibration/review-data.json`) is JSON with a different schema, so the detector flags it as stale and suggests manual cleanup.

**Expected**: Calibration data in `.copilot-tracking/calibration/` should be excluded from stale-artifact detection. It is persistent repo-specific data, not per-issue tracking state.

### 2. Cleanup check runs multiple times in a pipeline

The session startup check in `copilot-instructions.md` instructs every agent to run the cleanup detector at the start of every new conversation. In a multi-agent pipeline (for example, Code-Conductor → Experience-Owner → Code-Critic), each subagent invocation may trigger the check redundantly, producing duplicate cleanup prompts or wasting terminal round-trips.

**Expected**: The cleanup check should run at most once per pipeline session.

## Customer Experience

- **Surface**: Internal workflow — contributor session startup, issue pipeline handoff, and cleanup prompt flow
- **CE Gate**: Workflow validation should exist even though there is no external product surface

### Customer Problem Statement

Contributors and maintainers use the session startup check to keep their workspace clean after merges without thinking about internal housekeeping. Today that workflow breaks trust in two ways: it warns about calibration data that should persist, and it may repeat the same cleanup check multiple times inside one orchestration pipeline. From the contributor's perspective, the tool is creating noise instead of confidently helping them start work.

A good outcome is that the startup check only speaks up when there is real cleanup to do, never treats long-lived calibration data as disposable issue state, and runs once per workflow entry so multi-agent orchestration feels intentional rather than repetitive.

### User Segments

- **Primary**: Repository maintainers and contributors starting a new Copilot session after merging prior work
- **Secondary**: Users invoking hub-mode or multi-agent pipelines who expect one coherent startup experience, not repeated prompts from each subagent
- **Tertiary**: Workflow authors relying on calibration history to remain intact across sessions so review-pipeline quality signals are preserved

### User Journeys

**Current journey**

A contributor opens a new session. The startup detector scans `.copilot-tracking/`, misreads calibration JSON as an unknown stale artifact, and may suggest cleanup that should never happen. In multi-agent flows, the same contributor can also experience repeated detector executions and duplicate prompts as control passes between agents.

**Target journey**

A contributor opens a new session and sees one of two clean outcomes: either no prompt at all because only persistent calibration data is present, or one accurate cleanup prompt because real stale branch or per-issue tracking artifacts exist. If the work expands into a multi-agent pipeline, the startup check has already run and is not repeated.

**Edge journeys**

- A contributor with only `.copilot-tracking/calibration/review-data.json` present sees no stale-artifact warning
- A contributor with both real stale issue tracking files and calibration data sees cleanup guidance only for the stale issue artifacts
- A contributor entering via a top-level orchestrator sees the cleanup check once even if the workflow later invokes multiple subagents

### Scenarios

1. **Functional**: Start a new session with only `.copilot-tracking/calibration/review-data.json` present and verify the detector returns no stale-artifact cleanup prompt.
2. **Functional**: Start a new session with a real stale issue tracking file plus calibration data and verify the prompt references only the stale issue artifact, not calibration storage.
3. **Functional**: Invoke a multi-agent pipeline from a top-level entry point and verify the cleanup detector executes at most once for the whole pipeline session.
4. **Intent**: Verify the startup experience communicates a single clear message only when the contributor must take action, preserving trust that prompts map to real cleanup work.

### Named Design Decisions (Customer-Outcome Framing)

- **D1 — Persistent Learning Data Stays Silent**
  - Owner: Experience-Owner
  - Customer outcome: Calibration history is treated as durable repo memory, not as disposable per-issue clutter, so contributors do not get false cleanup alarms.
- **D2 — One Startup Check Per Workflow Entry**
  - Owner: Experience-Owner
  - Customer outcome: Starting a session or pipeline feels like one coherent interaction, not a chain of repeated housekeeping prompts.
- **D3 — Cleanup Prompts Only for Actionable State**
  - Owner: Experience-Owner
  - Customer outcome: When the system interrupts the contributor, the prompt is specific, correct, and worth acting on.

### Customer Surface Identification

- **Surface type**: Internal workflow / CLI-and-instruction surface
- **Primary surface**: Session startup instruction invoking `.github/scripts/session-cleanup-detector.ps1`
- **Supporting surfaces**: Cleanup prompt text, `.copilot-tracking/` directory semantics, multi-agent orchestration entry points

### Design Intent Reference

The startup cleanup experience should feel quiet, trustworthy, and singular: it should preserve long-lived workflow data, detect only true cleanup work, and avoid re-running the same housekeeping logic as the user moves through one orchestration pipeline.

### CE Gate Readiness Assessment

1. **Surface type**: Internal workflow validated through script execution and orchestration flow, not a browser UI
2. **Tool availability**: Terminal and repository scripts are sufficient; GitHub issue body and local workflow state provide the needed context
3. **Manual fallback**: If full pipeline orchestration is not runnable, validate with direct script invocation plus a controlled top-level agent entry and confirm prompt behavior from outputs/logs
4. **Draft scenarios**: The four scenarios above should be used later as workflow-experience validation even though there is no external end-user surface

### Technical Summary

The design separates two concerns that are currently conflated:

1. **What counts as stale issue-scoped tracking state**
   - The detector should only evaluate issue-scoped tracking files for stale-artifact cleanup.
   - `.copilot-tracking/calibration/**` is persistent repo-scoped calibration data and is excluded from stale-artifact detection.

2. **When the automatic startup check should run**
   - The detector script remains a pure detector.
   - Automatic run-once behavior belongs in the startup instruction layer, which records a session-memory marker after the first startup check for the current conversation and causes later agent hops in the same pipeline session to skip the automatic check.

This preserves a clean contract: the detector decides whether actionable cleanup exists, while the startup instructions decide whether the detector should be auto-invoked again in the current session.

### Design Decisions

- **TD1 — Classify calibration as persistent path-scoped data**
  - Choice: Treat `.copilot-tracking/calibration/**` as a persistent subtree and exclude it from the stale-artifact scan and from the detector's `unknown` bucket.
  - Rationale: `tracking-format.instructions.md` already defines calibration as persistent cache, so the detector should align to that semantic boundary instead of inferring cleanup eligibility from `issue_id` frontmatter.
  - Consequence: A workspace containing only calibration data produces `{}` from the detector and no cleanup prompt.

- **TD2 — Keep the detector script stateless and focused**
  - Choice: Do not embed pipeline-session run-once state inside `session-cleanup-detector.ps1`.
  - Rationale: The detector's job is classification and cleanup-context generation. Conversation-scoped orchestration belongs in the instruction layer, not inside a file-system detector script that has no natural concept of a pipeline session.
  - Consequence: The detector remains directly runnable from the terminal for manual verification or explicit troubleshooting.

- **TD3 — Add a conversation-scoped run-once marker in session memory**
  - Choice: The startup instructions write a small marker under `/memories/session/` after the first automatic startup check in a conversation. Later agents in the same pipeline session read that marker and skip the automatic detector invocation.
  - Rationale: Session memory matches the desired lifetime exactly: shared across the current conversation, cleared automatically when the session ends, and already used by this workflow for durable intra-session coordination.
  - Consequence: Automatic startup detection runs once per pipeline session without leaking state into future conversations or onto disk.

- **TD4 — Fail open on run-once guard errors**
  - Choice: If session-memory lookup or write fails, agents fall back to the current behavior and run the detector rather than silently suppressing it.
  - Rationale: A duplicate prompt is less harmful than missing a real cleanup warning. This preserves the existing safety property while the guard remains advisory.
  - Consequence: The design optimizes for trust and correctness over strict de-duplication when infrastructure is degraded.

- **TD5 — Scope the guard to automatic startup only**
  - Choice: The run-once marker suppresses only the instruction-driven automatic startup check. It does not block explicit user requests, direct script runs, or later implementation/test harness calls that intentionally invoke the detector.
  - Rationale: A once-per-session UX should not remove operator control.
  - Consequence: Manual troubleshooting and validation remain straightforward.

### User Experience

- Contributors starting a session with only calibration cache present see no cleanup prompt.
- Contributors starting a session with genuine stale issue-tracking artifacts still see one cleanup prompt with actionable guidance.
- Contributors moving through a multi-agent pipeline see the startup check at most once per conversation unless they explicitly re-run it.
- If the user declines cleanup, later subagents in the same conversation do not re-prompt; the check runs again in the next conversation.

### System Touchpoints

- `.github/scripts/session-cleanup-detector.ps1`
  - Add persistent-subtree classification so calibration data is excluded from stale-artifact detection.
- `.github/copilot-instructions.md`
  - Update the startup-check contract to use a session-memory run-once guard before auto-invoking the detector.
- `.github/instructions/session-startup.instructions.md`
  - Mirror the same run-once guidance and fallback behavior.
- Session memory under `/memories/session/`
  - Hold the conversation-scoped "startup check already performed" marker.
- Verification coverage
  - Extend detector tests for calibration exclusion.
  - Add instruction-level verification for run-once behavior across simulated multi-agent hops.

### Edge Cases

- **Calibration plus real stale issue files**: Only the issue-scoped stale files appear in cleanup guidance.
- **Calibration plus stale branch signal**: The stale-branch guidance still appears; calibration remains silent.
- **User declines cleanup**: The session-memory marker still records that startup checking already occurred, preventing duplicate prompts in the same conversation.
- **Session-memory read/write failure**: The detector still runs so no true cleanup need is missed.
- **Explicit manual re-run**: A user or test harness can still run the detector directly even if the automatic startup guard has already fired.
- **Future persistent tracking subtrees**: The implementation should express the exclusion as a named persistent-subtree classification, not a one-off filename special case, so later additions can extend the same model without redefining the detector contract.

### Testing Scope Decision

**Decision**: Unit + integration, no browser E2E.

**Rationale**: This is a cross-system workflow change spanning detector logic, instruction behavior, and session-memory coordination, but it does not introduce a browser surface or application UI.

#### Unit / script-level

1. Detector with only `.copilot-tracking/calibration/review-data.json` returns `{}`.
2. Detector with calibration data plus a real issue-tracking file reports only the issue-tracking artifact.
3. Detector with calibration data plus stale-branch conditions still emits the stale-branch context.

#### Integration / workflow-level

1. `startup instructions + session memory + detector` → first agent in a conversation runs the detector and records the run-once marker.
2. `subsequent agent hop + existing session marker` → automatic detector execution is skipped.
3. `new conversation + empty session memory` → automatic detector execution runs again.
4. `session-memory failure + startup instructions` → detector still runs, preserving current cleanup detection behavior.
5. `explicit manual detector invocation` → detector runs even if the startup marker already exists.

### Acceptance Criteria

- [ ] Automatic startup detection ignores `.copilot-tracking/calibration/**` when deciding whether stale artifacts exist.
- [ ] Calibration data is not listed in the detector's `unknown` stale-artifact output.
- [ ] Real stale issue-tracking artifacts are still detected when calibration data is present.
- [ ] Automatic startup detection runs at most once per conversation across a multi-agent pipeline session.
- [ ] A user who declines cleanup is not re-prompted by later agent hops in the same conversation.
- [ ] If the run-once guard cannot read or write session memory, cleanup detection still runs rather than being silently skipped.
- [ ] Manual or explicit detector runs remain possible after the automatic startup check has already occurred.
- [ ] Verification coverage exists for calibration exclusion and run-once behavior.

### Rejected Alternatives

- **Schema-based calibration exception**
  - Rejected because it ties cleanup classification to the current JSON shape and keeps the incorrect assumption that all `.copilot-tracking/` files are issue-scoped unless proven otherwise.

- **Top-level-agents-only policy**
  - Rejected because it depends on invocation mode always being correctly detectable and propagated across every subagent path. The run-once requirement is better enforced with conversation-scoped state than with convention alone.

- **Shell or local-file sentinel**
  - Rejected because state could leak across shells or future conversations, suppressing legitimate startup checks beyond the intended session boundary.

- **Broad user-configurable persistent-subdir registry**
  - Rejected for this issue because it adds a wider configuration surface than needed. The design still uses an explicit persistent-subtree concept so future persistent paths can extend the model deliberately if needed.

### Adversarial Design Challenge Summary

Three prosecution themes were applied to the chosen design: feasibility/risk, scope/completeness, and product-alignment. The merged challenge ledger and dispositions are:

1. **Risk: path exclusion could become an ad hoc one-off**
   - Disposition: incorporated.
   - Response: the design explicitly frames calibration as the first member of a persistent-subtree classification, not as a filename special case.

2. **Risk: run-once suppression could hide a real cleanup need if session memory fails**
   - Disposition: incorporated.
   - Response: the design fails open and runs the detector whenever the session-memory guard cannot be read or written.

3. **Product-alignment concern: once-per-session behavior could block intentional troubleshooting reruns**
   - Disposition: incorporated.
   - Response: the guard applies only to automatic startup execution; explicit manual runs remain allowed.

4. **Scope concern: declining cleanup once could still trigger repeated prompts later in the same conversation unless the marker records completion of the check, not completion of cleanup**
   - Disposition: incorporated.
   - Response: the run-once marker records that the startup check already occurred, regardless of whether the user ran cleanup.

### PR DESCRIPTION
**Pull request by Grimblaz**
## Summary

Implements issue #185 by separating persistent calibration state from stale cleanup detection and by documenting a conversation-scoped run-once guard for the automatic startup detector.

## Changed Files

- .github/copilot-instructions.md
- .github/instructions/session-startup.instructions.md
- .github/scripts/Tests/session-cleanup-detector.Tests.ps1
- .github/scripts/Tests/session-startup-wording-contract.Tests.ps1
- .github/scripts/session-cleanup-detector.ps1
- .github/scripts/test-session-cleanup-detector.ps1
- CONTRIBUTING.md
- CUSTOMIZATION.md
- Documents/Design/session-hooks.md

## Validation Evidence

- `pwsh -NoProfile -NonInteractive -Command "Invoke-Pester .github/scripts/Tests/ -Output Minimal"` → 43 passed, 0 failed
- `pwsh -NoProfile -NonInteractive -File .github/scripts/test-session-cleanup-detector.ps1` → 14 passed, 0 failed
- Structural checks → `{"PlanArchitect":0,"Janitor":0,"IssueDesigner":0,"WorkflowTemplate":0,"SkillDescriptionUse":0,"SkillDescriptionNoUseFor":0,"SkillGotchas":0}`
- `markdownlint-cli2 --fix "**/*.md"` was run as required; repo-wide markdownlint still reports pre-existing unrelated violations outside this change set

## CE Gate

✅ CE Gate passed — intent match: partial

Scenarios exercised:
- Calibration-only startup remains silent
- Calibration plus stale issue artifact reports only actionable cleanup state
- Run-once behavior verified through instruction-contract and session-marker evidence
- Prompt wording remains quiet when no action is needed and specific when cleanup is real

Process-Review: no systemic gap found

## Adversarial Review Scores

| Stage | Prosecutor | Defense | Judge rulings |
| --- | --- | --- | --- |
| Code Review | 5 pts (0 sustained) | 5 pts (0 disproved, 1 rejected) | 1 ruling |
| CE Review | 0 pts (0 sustained) | 0 pts (1 disproved, 0 rejected) | 1 ruling |
| Post-fix Review | ⏭️ N/A | ⏭️ N/A | ⏭️ N/A |

## Pipeline Metrics

Closes #185

### CODE REVIEW

File: .github/scripts/session-cleanup-detector.ps1 [hunk 1]
```diff
@@ -33,6 +33,42 @@ function Write-NoOp {
...
+
+        [Parameter(Mandatory)]
+        [string[]]$PersistentSubtrees
+    )
+
+    $filePath = [System.IO.Path]::GetFullPath($File.FullName)
+    $relativePath = [System.IO.Path]::GetRelativePath($trackingRootPath, $filePath).Replace('\', '/')
+
+    foreach ($subtree in $PersistentSubtrees) {
+        $normalizedSubtree = $subtree.Trim('/').Replace('\', '/')
```
[the remark this answers is not in the document]
Comment (Grimblaz) [reply .github/scripts/session-cleanup-detector.ps1]: Reviewed in GitHub intake and not accepted as a defect. The current implementation only uses the fixed literal subtree 'calibration', so the suggested whitespace trim is a hardening idea rather than evidence of a present failure mode in this PR.
